Core/Core/Views/HorizontalMenuViewController.swift

```diff
@@ -0,0 +1,278 @@
+    }
+
+    func setupUnderline() {
+        underlineView = UIView()
+        guard let underlineView = underlineView else { return }
+        view.addSubview(underlineView)
+        let w = view.bounds.size.width / CGFloat(itemCount)
+        underlineWidthConstraint = underlineView.addConstraintsWithVFL("H:[view(w)]", metrics: ["w": w])?.first
+        underlineLeftConstraint = underlineView.addConstraintsWithVFL("H:|[view]")?.first
+        underlineView.addConstraintsWithVFL("V:[view(2)]")
+        let bottoms = NSLayoutConstraint(item: underlineView, attribute: .bottom, relatedBy: .equal, toItem: menu, attribute: .bottom, multiplier: 1.0, constant: -1.1)
+        view.addConstraint(bottoms)
+        underlineView.backgroundColor = delegate?.menuItemSelectedColor ?? UIColor.blue
+    }
+
+    func setupBottomBorder() {
+        bottomBorder = UIView()
+        guard let bottomBorder = bottomBorder, let underlineView = underlineView else { return }
+        bottomBorder.backgroundColor = UIColor.named(.borderDark)
+        view.insertSubview(bottomBorder, belowSubview: underlineView)
+        bottomBorder.pinToLeftAndRightOfSuperview()
+        let bottoms = NSLayoutConstraint(item: bottomBorder, attribute: .bottom, relatedBy: .equal, toItem: menu, attribute: .bottom, multiplier: 1.0, constant: -1)
+        view.addConstraint(bottoms)
+        bottomBorder.addConstraintsWithVFL("V:[view(\(1.0 / UIScreen.main.scale))]")
+    }
+}
+
+extension HorizontalMenuViewController: UICollectionViewDataSource, UICollectionViewDelegate, UICollectionViewDelegateFlowLayout {
+    public func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return itemCount
+    }
+
+    public func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        if collectionView == menu {
+            let cell: MenuCell = collectionView.dequeue(for: indexPath)
+            cell.title?.text = delegate?.menuItemTitle(at: indexPath)
+            cell.title?.font = delegate?.menuItemFont
+            cell.title?.textColor = delegate?.menuItemDefaultColor
+            cell.selectionColor = delegate?.menuItemSelectedColor
+            cell.isAccessibilityElement = true
+            cell.accessibilityTraits = [.button, .header]
+            cell.accessibilityIdentifier = delegate?.accessibilityIdentifier(at: indexPath)
+            cell.accessibilityLabel = cell.title?.text
+            if indexPath == selectedIndexPath {
+                cell.isSelected = true
+                cell.accessibilityTraits.insert(.selected)
+            }
+
+            return cell
+        } else {
+            let identifier = "child_\(indexPath.item)"
+            let cell = collectionView.dequeueReusableCell(withReuseIdentifier: identifier, for: indexPath)
+            guard let vc = delegate?.viewControllers[indexPath.item] else { fatalError("\(#function) delegate nil") }
```

Comment:
Can this be an `if let` instead and don't `fatalError`? I like the `assert` above because it won't go into production but this could.